PLBenchmarks/edges.py

```diff
@@ -0,0 +1,174 @@
+                err0 = item._data[("DerivedMeasurement", "e_dg")]
+            if key == "lig_" + str(self._data[1]):
+                l1 = item._data["ROMol"][0]
+                s1 = item._data["smiles"]
+                dg1 = item._data[("DerivedMeasurement", "dg")]
+                err1 = item._data[("DerivedMeasurement", "e_dg")]
+        self._data["Mol1"] = l0
+        self._data["Mol2"] = l1
+        self._data["Smiles1"] = s0
+        self._data["Smiles2"] = s1
+        self._data["exp. DeltaG [kcal/mol]"] = round(dg1 - dg0, 2)
+        self._data["exp. Error [kcal/mol]"] = round(
+            np.sqrt(np.power(err0, 2.0) + np.power(err1, 2.0)), 2
+        )
+
+    def getDF(self, columns=None):
+        """
+        Access the edge data as a :py:class:`pandas.DataFrame`
+
+        :param cols: list of columns which should be returned in the :py:class:`pandas.DataFrame`
+        :return: :py:class:`pandas.DataFrame`
+        """
+        if columns:
+            return self._data[columns]
+        else:
+            return self._data
+
+    def getDict(self):
+        """
+        Access the edge data as a :py:class:`dict` which contains the name of the edge as key and the names of the two ligands as :py:class:`list`.
+
+        :return: :py:class:`dict`
+        """
+        return {
+            f"edge_{self._data[0]}_{self._data[1]}": [
+                f"lig_{self._data[0]}",
+                f"lig_{self._data[1]}",
+            ]
+        }
+
+    def getName(self):
+        """
+        Access the name of the edge.
+
+        :return: name as string
+        """
+        if self._name is not None:
+            return self._name
+        else:
+            return f"edge_{self._data[0]}_{self._data[1]}"
+
+
+class edgeSet(dict):
+    """
+    Class inherited from dict to store all available edges of one target.
+    """
+
+    def __init__(self, target, *arg, **kw):
+        """
+        Initializes edgeSet class
+
+        :param target: string name of target
+        :param arg: arguments for :py:class:`dict` (base class)
+        :param kw: keywords for :py:class:`dict` (base class)
+        """
+        super(edgeSet, self).__init__(*arg, **kw)
+        tp = PLBenchmarks.targets.getTargetDataPath(target)
+        ligs = PLBenchmarks.ligands.ligandSet(target)
+        file = open(os.path.join(tp, "edges.yml"))
+        data = yaml.full_load_all(file)
+        for d in data:
+            e = edge(d)
+            e.addLigData(ligs)
+            self[e.getName()] = e
+        file.close()
+
+    def getEdge(self, name):
+        """
+        Accesses one edge of the :py:class:`PLBenchmarks.edges.edgeSet`
+
+        :param name: string name of the edge
+        :return: :py:class:`PLBenchmarks:edges:edge` class
+        """
+        for key in self.keys():
+            if key == name:
+                return self[key]
+                break
+        else:
+            raise ValueError(f"Edge {name} not part of set.")
```

Comment:
```suggestion
        if name not in self.keys():
            raise ValueError(f"Edge {name} not part of set.")

        for key in self.keys():
            if key == name:
                return self[key]
```
Checking for membership in a list-like is fast, so you can validate the input before iterating through the keys. Also don't need a `break` since `return` exits out of the function

Reply:
Made it even simpler in commit 
https://github.com/openforcefield/PLBenchmarks/pull/3/commits/bececbd14a0758487ef7d28ccadc36b09ad650b4